pkg/cache/v3/resource.go

```diff
@@ -212,42 +210,18 @@ func getListenerReferences(src *listener.Listener, out map[resource.Type]map[str
 				continue
 			}
 
-			routeSpecifier := config.RouteSpecifier
-			switch r := routeSpecifier.(type) {
-			case *hcm.HttpConnectionManager_Rds:
-				if r == nil || r.Rds == nil {
-					break
-				}
-
-				routes[r.Rds.RouteConfigName] = true
-
-			case *hcm.HttpConnectionManager_ScopedRoutes:
-				if r == nil || r.ScopedRoutes == nil {
-					break
-				}
-
-				// If we are using SRDS add the scoped route
-				// resource.
-				if r.ScopedRoutes.GetScopedRds() != nil {
-					scopedRoutes[r.ScopedRoutes.Name] = true
-				}
-
-				// If the scoped route mapping is embedded, add
-				// the referenced route resource names.
-				for _, s := range r.ScopedRoutes.GetScopedRouteConfigurationsList().GetScopedRouteConfigurations() {
-					routes[s.RouteConfigurationName] = true
-				}
+			// If we are using RDS, add the referenced the route name.
+			if name := config.GetRds().GetRouteConfigName(); name != "" {
+				routes[name] = true
 			}
-		}
-	}
 
-	if len(scopedRoutes) > 0 {
-		if _, ok := out[resource.ScopedRouteType]; !ok {
-			out[resource.ScopedRouteType] = map[string]bool{}
+			// If the scoped route mapping is embedded, add the referenced route resource names.
+			for _, s := range config.GetScopedRoutes().GetScopedRouteConfigurationsList().GetScopedRouteConfigurations() {
```

Comment:
Should we be nil checking each call? 

Is there potential for GetScopedRoutes() to return nil? 
Is there potential for GetScopedRouteConfigurationsList() to return nil?

Reply:
protobuf generates the `Get` accessors to gracefully handle nil objects. That's convenient for this kind of code because if any of the fields in the chain of references is nil, we will just end up ranging over a nil slice (which is a no-op).